templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -499,25 +556,36 @@ contents:
       echo "${physical_interface}"
     }
 
-    # Accepts parameters $iface, $iface_default_hint_file
-    # Finds the default interface. If the default interface is br-ex, use that and return.
+    # Accepts parameters $iface, $iface_default_hint_file, $ip_hint_file
+    # Finds the default interface.
+    # Reading $ip_hint_file and searching for interface that matches this ip.
+    # In case $ip_hint_file was not set use flow below.
+    # If the default interface is br-ex, use that and return.
```

Comment:
nit: I propose this instead
```
# Accepts parameters $iface, $iface_default_hint_file, $ip_hint_file
# Finds the nodeip interface from the interface that matches the ip address in $ip_hint_file.
# Otherwise fallbacks to a previously used interface or to the default interface.ç
# Never use the interface that is provided inside extra_bridge_file for br-ex1.
# Never use br-ex1.
# Read $ip_hint_file and return the interface that matches this ip.  Otherwise:
# If the default interface is br-ex, use that and return.
# If the default interface is not br-ex:
# Check if there is a valid hint inside iface_default_hint_file. If so, use that hint.
# If there is no valid hint, use the default interface that we found during the step
# earlier. Write the default interface to the hint file.
```

Reply:
Reading this I realized that this might lead to a change in the interface from what was previously used, which is what iface_default_hint_file is trying to prevent.

This was done because of bug https://bugzilla.redhat.com/show_bug.cgi?id=2057160 but I fail to see any actual problem documented in the bug.

@rbbratta @andreaskaris are you aware of what actual problems did changing the interface used actually bring up, if any?

If we are not sure about this, would it be a problem to honor iface_default_hint_file above ip_hint_file? Or are we actively seeking support to move to another interface after an upgrade (or after a reboot)?

Reply:
We should support moving IMO. We need to be aligned with nodeip service this is actual idea of it. Though it should always be the same interface.

Reply:
We should be coupled with nodeip service another way if it will set ip that is not matching interface it will cause very strange failures, those that we are trying to fix with this PR actually. Ovs should not decide from now, it should take whatever nodeip decide to be right by setting node-ip file and ovs just takes interface that matches it

Reply:
I understand that but I am not sure right now if this caused any problems in ovn-kubernetes, I guess not.
Had you have the chance to actually try out what happens when we actually move from another interface?
I don't think we have any actual verification of this.

Reply:
Interface was changed as expected but it didn't actually look on what is going on with cluster itself to say the truth. Though chance if getting another interface for the same IP is really low

Reply:
@jcaamano in [2057160](https://bugzilla.redhat.com/show_bug.cgi?id=2057160) the different interfaces were on different subnets, so I assume they didn't have connectivity on the other subnets.

```
Feb 22 17:54:43 master-0 configure-ovs.sh[2310]:     inet 192.168.111.20/24 brd 192.168.111.255 scope global dynamic noprefixroute br-ex
Feb 22 17:54:43 master-0 configure-ovs.sh[2310]: 192.168.111.0/24 dev br-ex proto kernel scope link src 192.168.111.20 metric 49
Feb 22 18:33:11 master-0 configure-ovs.sh[2775]:     inet 192.168.111.20/24 brd 192.168.111.255 scope global dynamic noprefixroute br-ex
Feb 22 18:33:11 master-0 configure-ovs.sh[2775]: 192.168.111.0/24 dev br-ex proto kernel scope link src 192.168.111.20 metric 49
Feb 22 18:33:15 master-0 configure-ovs.sh[2775]:     inet 192.168.221.20/24 brd 192.168.221.255 scope global dynamic noprefixroute br-ex
Feb 22 18:33:15 master-0 configure-ovs.sh[2775]: 192.168.221.0/24 dev br-ex proto kernel scope link src 192.168.221.20 metric 49
```

The issue would also be that moving interfaces changes IP.  Changing IP breaks Etcd Certs on masters.

```
EtcdCertSignerControllerDegraded: [x509: certificate is valid for 10.30.1.4, not 2101::4, x509: certificate is valid for ::1, 10.30.1.4, 127.0.0.1, ::1, not 2101::4] here the certificate is valid for IPV4
```

I don't think we ran into a case where interface changed but IP didn't.

 [2057160](https://bugzilla.redhat.com/show_bug.cgi?id=2057160) is tricky in that it is IPI OSP with `additionalNetworkIDs`.  OSP `additionalNetworkIDs` is the only case of multi-homed IPI that I know of. 


We might be able to test with a bonding scenario, if both slaves have the same static IP DHCP mapping and `bond0` fails to activate somehow on boot, NM should fall back to just using one of the slaves, soIP might stay the same but interface change on reboot from `bond0` to `ens192` or some such.